app/controllers/api/v1/puzzles_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Api::V1::PuzzlesController < ApplicationController
+
+  def index
+
+      zip_code = params[:zip_code]
+
+      users = User.where(zip_code: zip_code) if zip_code.present?
```

Comment:
This logic should probably be a method in the models and only called on here in the controller. But again, to meet MVP within out time limits let's just roll with it for now. Nice work.